web/template.py

```diff
@@ -54,6 +56,41 @@
 from collections.abc import MutableMapping
 
 
+class BetterIter:
+    """Iterator like object with 2 support for 2 look aheads."""
+
+    def __init__(self, items):
+        self.iteritems = iter(items)
+        self.items = []
+        self.position = 0
+        self.current_item = None
+
+    def lookahead(self):
+        if len(self.items) <= self.position:
+            self.items.append(self._next())
+        return self.items[self.position]
+
+    def _next(self):
+        try:
+            return next(self.iteritems)
+        except StopIteration:
+            return None
+
+    def lookahead2(self):
+        if len(self.items) <= self.position + 1:
+            self.items.append(self._next())
+        return self.items[self.position + 1]
+
+    def __next__(self):
+        self.current_item = self.lookahead()
+        self.position += 1
+        return self.current_item
```

Comment:
It doesn't look like `lookahead2()` is used at all, so this could be replaced by [`more-itertools.peekable`](https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.peekable) but I don't feel strongly about it (and it does introduce another dependency)

Reply:
Oops, scratch that, lookahead2 actually does appear to be used and I missed it. Sorry!

Reply:
But `peekable` can still be used with `peekable[1]` replacing `lookahead2()`. See the commit referenced below.